Controllers/BookController.php

```diff
@@ -32,16 +33,35 @@ public function bookPageAction($id)
 
   public function deletePage($id)
   {
-    if (!empty($_POST )) {
+    if (!empty($_POST)) {
       $bookModel = new BookModel();
       $bookModel->deleteBook($id);
       set_message('Deleted successfully', 'status');
       redirect("/admin");
-
-    }
-    else {
+    } else {
       $this->content = $this->render('/views/books/delete_book_page.tpl.php');
       $this->renderLayout('/views/layouts/basic.tpl.php');
     }
   }
+
+  /**
+   * Callback for /admin/book/edit/%id route.
+   */
+  public function editPageAction($id)
+  {
+    $bookModel = new BookModel();
+    if (!empty($_POST)) {
+      $_POST["buy_link"] = $_POST["buy-link"];
+      $bookModel->update(new BookEntity($_POST));
+      if (empty(get_messages()))
+        set_message('Book successfully updated!', 'status');
+    }
+    $book = $bookModel->get($id);
+    $this->title = $book->getTitle();
+    $messages = render_messages(get_messages());
```

Comment:
I think this is not needed. As far as I remember, we have already something in BasicController that catches all messages.